File: C++/Level1/명예의 전당.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> answer;
    priority_queue<int, vector<int>, greater<int>> honor; // 명예의 전당
    for (int s : score) {
        honor.push(s);
        if (k < honor.size()) 
            honor.pop();
        answer.push_back(honor.top());
    }

    return answer;
}

File: programmers/Java/Level1/명예의 전당.java
import java.util.PriorityQueue;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        PriorityQueue<Integer> honor = new PriorityQueue<Integer>();
        for (int i = 0; i < score.length; i++) {
            honor.add(score[i]);
            if (k < honor.size())
                honor.poll();
            answer[i] = honor.peek();
        }
        return answer;
    }
}